Developer-instructions.md

```diff
@@ -0,0 +1,253 @@
+# Developer instructions
+
+## Recommended setup
+
+Our recommended setup for working on the library is by using vscode on Linux, MacOS, or WSL, with
+the coq-lsp and ocaml extensions. We will now describe more extensively how to achieve this setup.
+
+### Installation of opam
+
+A more extensive description can be found [here](https://ocaml.org/docs/installing-ocaml).
+This is a short description.
+
+On some versions of Linux and WSL, you can install opam with
+```
+apt-get install opam
+```
+On MacOs, you can either use homebrew
+```
+brew install gpatch
+brew install opam
+```
+or macports
+```
+port install opam
+```
+
+### Creating the desired opam environment: simplified version
+When we develop, we need to be aware of the version of Coq we develop for.
+Most development is done against the main branch, and then the following setup will suffice.
+
+```
+opam init
+eval $(opam env)
+opam install coq-lsp.2.0.0+8.17
```

Comment:
How often does this version change? If it changes often maybe note that above this command, if there's a way we can reliably install some latest version, maybe give that command instead. Maybe the edit below is enough?

Reply:
I'll go with your suggestion below. We may indeed want to think about how to get all version information up to date.